server/server.go

```diff
@@ -357,6 +359,27 @@ func (w *otherWorker) Do(job interface{}) interface{} {
 		}
 		return w.s.makeRespondResponse(req, codec.response, requestBegin)
 
+	case protocol.OpEd25519Sign:
+		keyLoadBegin := time.Now()
+		key, err := w.s.keys.Get(&pkt.Operation)
+		if err != nil {
+			log.Errorf("failed to load key with sni=%s ip=%s ski=%v: %v", pkt.Operation.SNI, pkt.Operation.ServerIP, pkt.Operation.SKI, err)
+			return w.s.makeErrResponse(req, protocol.ErrInternal, requestBegin)
+		} else if key == nil {
+			log.Errorf("failed to load key with sni=%s ip=%s ski=%v: %v", pkt.Operation.SNI, pkt.Operation.ServerIP, pkt.Operation.SKI, protocol.ErrKeyNotFound)
+			return w.s.makeErrResponse(req, protocol.ErrKeyNotFound, requestBegin)
+		}
+		w.s.stats.logKeyLoadDuration(keyLoadBegin)
+
+		ed25519Key, ok := key.(ed25519.PrivateKey)
```

Comment:
You need a contingency for key not being of type ed25519.PrivateKey.

Reply:
I believe that's handled on line 375. If ok==true, then on line 380, the first argument to ed25519.Sign() is the right type.

Reply:
I mean that you need to handle the case where keys.Get() returns a 'keyless' (or some other abstracted) private key. A la line 400 in OpRSADecrypt.
```
if ok {
    // type-specific things
} else {
    // delegate type-specific things to the interface implementation
}
```

Reply:
Done.